hummingbot/strategy/pure_market_making/pure_market_making_v2.pyx

```diff
@@ -295,10 +318,40 @@ cdef class PureMarketMakingStrategyV2(StrategyBase):
                                                                                    market_info,
                                                                                    market_info_to_active_orders.get(market_info, []),
                                                                                    orders_proposal)
+                filtered_proposal = self.c_filter_orders_proposal_for_takers(market_info, filtered_proposal)
                 self.c_execute_orders_proposal(market_info, filtered_proposal)
         finally:
             self._last_timestamp = timestamp
 
+    # To filter out any orders that are going to be taker orders, i.e. buy order price higher than first ask
+    # and sell order price lower than first bid on the order book.
+    cdef object c_filter_orders_proposal_for_takers(self, object market_info, object orders_proposal):
+        cdef :
+            list buy_prices = []
+            list buy_sizes = []
+            list sell_prices = []
+            list sell_sizes = []
+        if orders_proposal.buy_order_sizes[0] > 0:
+            for i in range(0, len(orders_proposal.buy_order_prices)):
+                first_ask = market_info.get_price(True)
```

Comment:
While this works... it's slow performance-wise.

 1. We're in a `c_` function... we should minimize the number of pure Python calls if we can help it. So instead of `market_info.get_price()`, this should be `market.c_get_price()` instead. The `market` object should be stored as a `cdef` allocated type in the beginning so we don't have to go through the tuple attribute traversal.
 2. It doesn't need to be called for each iteration. It should be cached before the `for ...` loop.

Reply:
Ok, I'll be more careful with cython function calls.

Reply:
You mean having market object in OrderBookAssetPriceDelegate instead of market_info and use c_get_price instead of get_price? I think that makes sense, though after I made the change, my unit testing breaks, it complains 
` 'hummingsim.backtest.backtest_market.BacktestMarket' object has no attribute 'c_get_price'`